drivers/sensor/adi/adxl372/adxl372_stream.c

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) 2024 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor.h>
+
+#include "adxl372.h"
+
+LOG_MODULE_DECLARE(ADXL372, CONFIG_SENSOR_LOG_LEVEL);
+
+void adxl372_submit_stream(const struct device *dev, struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct sensor_read_config *cfg =
+			(const struct sensor_read_config *) iodev_sqe->sqe.iodev->data;
+	struct adxl372_data *data = (struct adxl372_data *)dev->data;
+	const struct adxl372_dev_config *cfg_372 = dev->config;
+	uint8_t int_mask = 0, status1, status2;
+	uint16_t fifo_entries = 0;
+
+	int rc = gpio_pin_interrupt_configure_dt(&cfg_372->interrupt,
+					      GPIO_INT_DISABLE);
+
+	if (rc < 0) {
+		return;
+	}
+
+	for (size_t i = 0; i < cfg->count; i++) {
+		if ((cfg->triggers[i].trigger == SENSOR_TRIG_FIFO_WATERMARK) ||
+			(cfg->triggers[i].trigger == SENSOR_TRIG_FIFO_FULL)) {
+			int_mask |= ADXL372_INT1_MAP_FIFO_FULL_MSK;
+		}
+	}
+
+	if (int_mask) {
+		rc = data->hw_tf->write_reg_mask(dev, ADXL372_INT1_MAP, int_mask, int_mask);
+		if (rc < 0) {
+			return;
+		}
+
+		/* Flush the FIFO by disabling it. Save current mode for after the reset. */
+		enum adxl372_fifo_mode current_fifo_mode = cfg_372->fifo_config.fifo_mode;
+
+		adxl372_configure_fifo(dev, ADXL372_FIFO_BYPASSED, cfg_372->fifo_config.fifo_format,
+				cfg_372->fifo_config.fifo_samples);
+		adxl372_configure_fifo(dev, current_fifo_mode, cfg_372->fifo_config.fifo_format,
+				cfg_372->fifo_config.fifo_samples);
+		adxl372_set_op_mode(dev, cfg_372->op_mode);
+	}
+
+	rc = adxl372_get_status(dev, &status1, &status2, &fifo_entries); /* Clear status */
+	if (rc < 0) {
+		return;
+	}
+
+	rc = gpio_pin_interrupt_configure_dt(&cfg_372->interrupt,
+					      GPIO_INT_EDGE_TO_ACTIVE);
+	if (rc < 0) {
+		return;
+	}
+
+	data->sqe = iodev_sqe;
+}
+
+static void adxl372_fifo_read_cb(struct rtio *rtio_ctx, const struct rtio_sqe *sqe, void *arg)
+{
+	const struct device *dev = (const struct device *)arg;
+	const struct adxl372_dev_config *cfg = (const struct adxl372_dev_config *) dev->config;
+	struct rtio_iodev_sqe *iodev_sqe = sqe->userdata;
+
+	rtio_iodev_sqe_ok(iodev_sqe, 0);
+
+	gpio_pin_interrupt_configure_dt(&cfg->interrupt, GPIO_INT_EDGE_TO_ACTIVE);
+}
+
+size_t adxl372_get_packet_size(const struct adxl372_dev_config *cfg)
+{
+	/* If one sample contains XYZ values. */
+	size_t packet_size = 6;
+
+	switch (cfg->fifo_config.fifo_format) {
+	case ADXL372_X_FIFO:
+	case ADXL372_Y_FIFO:
+	case ADXL372_Z_FIFO:
+		packet_size = 2;
+		break;
+
+	case ADXL372_XY_FIFO:
+	case ADXL372_XZ_FIFO:
+	case ADXL372_YZ_FIFO:
+		packet_size = 4;
+		break;
+
+	default:
```

Comment:
packet_size = 6 (in the default case) here would be clearer than initializing it above imho